src/box/lua/config/applier/roles.lua

```diff
@@ -57,10 +108,18 @@ local function apply(config)
             end
         end
         loaded[role_name] = role
-        table.insert(loaded_names_ordered, role_name)
+        if role.dependencies ~= nil and type(role.dependencies) ~= 'table' then
+            local err = 'Wrong type of role %s dependencies'
+            error(err:format(role_name), 0)
+        end
+        roles_dependencies[role_name] = {dependencies = role.dependencies or {}}
     end
 
+    -- Re-sorting of roles taking into account dependencies between them.
+    roles_ordered = resort_roles(roles_ordered, roles_dependencies)
```

Comment:
Nit: Can't we just pass `loaded` instead of `roles_dependencies`? (`ipairs()` in `add_role` should be updated with `or {}` in this case.)

Reply:
Done.